### ISSUE TITLE
[BUG] QNodes with `qml.kUpCCGSD` and `diff_method="parameter-shift` give incorrectly-sized gradients

### ISSUE THREAD

**Issue by isaacdevlugt** [PennyLaneAI/pennylane#2910]
### Expected behavior

Gradients of QNodes including the `qml.kUpCCGSD` operation using `"parameter-shift"` return an array whose shape is that of the differentiable parameters.

### Actual behavior

Gradients of QNodes including the `qml.kUpCCGSD` operation using `"parameter-shift"` return a scalar.

### Additional information

From this discussion post:

https://discuss.pennylane.ai/t/quantum-natural-gradient-is-bugged-with-kupccgsd/2081

@josh146 found the solution, which involves adding `grad_method=None` under this line: https://github.com/PennyLaneAI/pennylane/blob/37070078fe0a29de5f8d59a70e308d427105f970/pennylane/templates/subroutines/kupccgsd.py#L204

### Source code

```shell
import pennylane as qml
from pennylane import numpy as np
​
dev = qml.device("default.qubit", wires=4)
​
def cost(params):
    qml.kUpCCGSD(weights=params, wires=[0, 1, 2, 3], k=1, delta_sz=0, init_state=[0, 0, 1, 1])
    return qml.expval(qml.PauliZ(3))
​
​
params = np.ones((1, 6), requires_grad=True)
​
ps = qml.QNode(cost, dev, diff_method="parameter-shift")
print(qml.grad(ps)(params))
# Out: a scalar!
'''
4.440892098500626e-09
'''

# nothing wrong with backprop, but putting here for context

backprop = qml.QNode(cost, dev, diff_method="backprop")
print(qml.grad(backprop)(params))
# Out:
'''
[[-4.16333634e-17  3.19189120e-16 -4.16333634e-17 -1.80411242e-16
   1.44849410e-16 -2.22911967e-16]]
'''
```

### System information

```shell
Name: PennyLane
Version: 0.24.0
Summary: PennyLane is a Python quantum machine learning library by Xanadu Inc.
Home-page: https://github.com/XanaduAI/pennylane
Author: 
Author-email: 
License: Apache License 2.0
Location: /opt/conda/envs/pennylane/lib/python3.9/site-packages
Requires: appdirs, autograd, autoray, cachetools, networkx, numpy, pennylane-lightning, retworkx, scipy, semantic-version, toml
Required-by: PennyLane-Cirq, PennyLane-Lightning, PennyLane-qiskit, pennylane-qulacs, PennyLane-SF

Platform info:           Linux-5.4.196-108.356.amzn2.x86_64-x86_64-with-glibc2.31
Python version:          3.9.13
Numpy version:           1.23.1
Scipy version:           1.8.1
Installed devices:
- default.gaussian (PennyLane-0.24.0)
- default.mixed (PennyLane-0.24.0)
- default.qubit (PennyLane-0.24.0)
- default.qubit.autograd (PennyLane-0.24.0)
- default.qubit.jax (PennyLane-0.24.0)
- default.qubit.tf (PennyLane-0.24.0)
- default.qubit.torch (PennyLane-0.24.0)
- lightning.qubit (PennyLane-Lightning-0.24.0)
- qiskit.aer (PennyLane-qiskit-0.23.0)
- qiskit.basicaer (PennyLane-qiskit-0.23.0)
- qiskit.ibmq (PennyLane-qiskit-0.23.0)
- qiskit.ibmq.circuit_runner (PennyLane-qiskit-0.23.0)
- qiskit.ibmq.sampler (PennyLane-qiskit-0.23.0)
- cirq.mixedsimulator (PennyLane-Cirq-0.22.0)
- cirq.pasqal (PennyLane-Cirq-0.22.0)
- cirq.qsim (PennyLane-Cirq-0.22.0)
- cirq.qsimh (PennyLane-Cirq-0.22.0)
- cirq.simulator (PennyLane-Cirq-0.22.0)
- strawberryfields.fock (PennyLane-SF-0.20.1)
- strawberryfields.gaussian (PennyLane-SF-0.20.1)
- strawberryfields.gbs (PennyLane-SF-0.20.1)
- strawberryfields.remote (PennyLane-SF-0.20.1)
- strawberryfields.tf (PennyLane-SF-0.20.1)
- qulacs.simulator (pennylane-qulacs-0.16.0)
```

### Existing GitHub issues

- [X] I have searched existing GitHub issues to make sure the issue does not already exist.

**Comment by isaacdevlugt** [comment]
Working on fix. See https://github.com/PennyLaneAI/pennylane/pull/2913
